src/pybind/mgr/cephadm/module.py

```diff
@@ -1690,6 +1690,15 @@ def add_host(self, spec):
         self.log.info('Added host %s' % spec.hostname)
         return "Added host '{}'".format(spec.hostname)
 
+    def _add_host(self, spec):
+        # type: (HostSpec) -> str
+        completion = self.add_host(spec)
+        self.process([completion])
+        while completion.needs_result:
+            time.sleep(1)
+        orchestrator.raise_if_exception(completion)
+        return completion.result_str()
```

Comment:
I'd do it the other way around. 

1. rename `def add_host(self, spec: HostSpec):` to _add_host
2. add

```python
@async_completion
def add_host(self, spec):
    return self._add_host(spec)
```